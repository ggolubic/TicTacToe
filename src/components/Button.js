import styled, { css } from "styled-components";

const Button = styled.button`
  font-family: "Montserrat", sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  transition: 0.3s;
  cursor: pointer;
  outline: none;

  &:hover {
    color: white;
    background: #ddbc95;
    transition: 0.3s;
  }
  ${props =>
    props.new &&
    css`
      background: #ddbc95;
      color: white;
      &:hover {
        color: #ddbc95;
        background: white;
        transition: 0.3s;
      }
    `};
`;

export default Button;
