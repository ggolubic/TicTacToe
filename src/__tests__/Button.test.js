import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import "jest-styled-components";
import Button from "../components/Button";

Enzyme.configure({ adapter: new Adapter() });

describe("Button component", () => {
  it("renders without props correctly", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with props correctly", () => {
    const tree = renderer.create(<Button new />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render passed text", () => {
    const component = mount(<Button>"Thing"</Button>);
    expect(component.find("button").text()).toContain("Thing");
    component.unmount();
  });

  it("should call props.onClick when button is clicked", () => {
    const mockFunction = jest.fn();
    const component = mount(<Button onClick={mockFunction} />);
    // Test before event
    expect(mockFunction).not.toHaveBeenCalled();

    // simulate the click event
    component.find("button").simulate("click");

    // Test after event
    expect(mockFunction).toHaveBeenCalled();
    component.unmount();
  });
});
