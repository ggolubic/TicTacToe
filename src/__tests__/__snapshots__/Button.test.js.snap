// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders with props correctly 1`] = `
.c0 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
  background: #ddbc95;
  color: white;
}

.c0:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0:hover {
  color: #ddbc95;
  background: white;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

<button
  className="c0"
/>
`;

exports[`Button component renders without props correctly 1`] = `
.c0 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
}

.c0:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

<button
  className="c0"
/>
`;
